library/waku.nim

```diff
@@ -0,0 +1,399 @@
+                  some(Port(uint16(config.port)))
+                else:
+                  extTcpPort
+
+  let wakuFlags = CapabilitiesBitfield.init(
+        lightpush = false,
+        filter = false,
+        store = false,
+        relay = config.relay
+      )
+
+  let netConfigRes = NetConfig.init(
+      bindIp = listenAddr,
+      bindPort = Port(uint16(config.port)),
+      extIp = extIp,
+      extPort = extPort,
+      wakuFlags = some(wakuFlags))
+
+  if netConfigRes.isErr():
+    let msg = string("Error creating NetConfig: ") & $netConfigRes.error
+    jsonResp = errResp(msg)
+    return false
+
+  netConfig = netConfigRes.value
+
+  return true
+
+# WakuNode instance
+var node {.threadvar.}: WakuNode
+
+### End of not-exported components
+################################################################################
+
+################################################################################
+### Exported procs
+
+proc waku_new(config: ConfigNode,
+              jsonResp: var string): bool
+              {.dynlib, exportc.} =
+# Creates a new instance of the WakuNode.
+# Notice that the ConfigNode type is also exported and available for users.
+  var privateKey: PrivateKey
+  var netConfig = NetConfig.init(ValidIpAddress.init("127.0.0.1"), Port(60000'u16)).value
+  if not parseConfig(config,
+                     privateKey, netConfig,
+                     jsonResp):
+    return false
+
+  var builder = WakuNodeBuilder.init()
+  builder.withRng(crypto.newRng())
+  builder.withNodeKey(privateKey)
+  builder.withNetworkConfiguration(netConfig)
+  builder.withSwitchConfiguration(
+    maxConnections = some(50.int)
+  )
+
+  let wakuNodeRes = builder.build()
+  if wakuNodeRes.isErr():
+    let errorMsg = string("failed to create waku node instance: ") & wakuNodeRes.error
+    jsonResp = errResp(errorMsg)
+    return false
+
+  node = wakuNodeRes.value
+
+  if config.relay:
+    waitFor node.mountRelay()
+    node.peerManager.start()
+
+  return true
+
+proc waku_version(): cstring {.dynlib, exportc.} =
+  return wakuNode2VersionString
+
+proc waku_set_event_callback(callback: EventCallback) {.dynlib, exportc.} =
+  eventCallback = callback
+
+proc waku_content_topic(appName: cstring,
+                        appVersion: uint,
+                        contentTopicName: cstring,
+                        encoding: cstring,
+                        outContentTopic: var string) {.dynlib, exportc.} =
+  # https://rfc.vac.dev/spec/36/#extern-char-waku_content_topicchar-applicationname-unsigned-int-applicationversion-char-contenttopicname-char-encoding
+  outContentTopic = fmt"{appName}/{appVersion}/{contentTopicName}/{encoding}"
+
+proc waku_pubsub_topic(topicName: cstring, encoding: cstring, outPubsubTopic: var string) {.dynlib, exportc.} =
+  # https://rfc.vac.dev/spec/36/#extern-char-waku_pubsub_topicchar-name-char-encoding
+  outPubsubTopic = fmt"/waku/2/{topicName}/{encoding}"
```

Comment:
Note that for pubsub topics we've deprecated the `{encoding}` suffix (https://rfc.vac.dev/spec/23/#pubsub-topic-format).